content/docs/ops/runbook/rotating-cloudfoundry-diego.md

```diff
@@ -0,0 +1,206 @@
+---
+menu:
+  docs:
+    parent: runbook
+
+title: Rotating Secrets V - Cloud Foundry and Diego
+---
+
+## Introduction
+
+When rotating secrets for Cloud Foundry and Diego you will need to follow the
+proper steps in order to perform a rolling update to the platform. There are six
+deployments that need to be synchronized across two Bosh deployments. These
+deployments are focused on the Consul components for Cloud Foundry and Diego.
+
+Follow the following documentation to get familiar with which values are going
+to be replaced.
+
+- [Cloud Foundry AWS Stub](http://docs.cloudfoundry.org/deploying/aws/cf-stub.html)
+- [Security Configuration for Consul](http://docs.cloudfoundry.org/deploying/common/consul-security.html)
+- [Generating all the Cloud Foundry certificates](https://github.com/18F/cg-deploy-cf#how-to-generate-the-final-manifest)
+
+## Prepare a local directory for rotating secret files
+
+Create a directory you can use to generate all the various artifacts that need
+to be rotated for the secrets files. Create a `./tmp` directory inside the
+`cg-deploy-cf` repository. Delete this directory at the end of secrets rotation.
+
+## Dealing with multiple secret files
+
+The Cloud Foundry manifest is split into two parts, a `main.yml` and
+`external.yml`. Internal secrets can be rotated either all at once or on a
+rolling schedule of deployments. You can pull the information needed to decrypt
+these from the `cg-deploy-cf` pipeline. You can easily grab the information
+needed to [decrypt and encrypt secrets]({{< relref "docs/ops/secrets.md" >}}) from
+Concourse using the `fly-cli`.
+
+```sh
+# note the target name `fr` here is the output of `fly targets`
+$ fly -t fr get-pipeline --pipeline deploy-cf | \
+  spurce json | \
+  jq -r '.resources | select(.name | test("common-"))'
+```
+
+### Internal secrets (main.yml)
+
+1. Rotate all UAA user passwords with passwords matching the UAA restrictions
+   set in [the spec](https://github.com/18F/uaa-customized-boshrelease/) for our
+   `uaa-customized-boshrelease`.
```

Comment:
yeah the idea here is that since it's in a deployment, you won't get the _reject bad credentials bit_ until you deploy the thing, and it's not the best feedback loop. ➿